### ISSUE TITLE
Quota handled through script and UI is inconsistent [Help]

### ISSUE THREAD

**Issue by hakila** [Axway-API-Management-Plus/apimanager-swagger-promote#120]
**Swagger-Promote Version**1.6

**API-Manager and Service-Pack Version**7.7

**Question**
For an api, when I try to add default quota, it overrides if it exists and makes sure one entry is there when doing through script.
In the UI, we can add duplicate quotas. Is it possible to make ui also works the same way as script for quota.

**What I've tried so far**
tried quota 
```json
{
  "restrictions": [
    {
      "method": "*",
      "type": "throttlemb",
      "config": {
        "period": "hour",
        "mb": 27,
        "per": 1
      }
    }
  ]
}
```
for the first time
Second time, I used
```json
 {
  "restrictions": [
    {
      "method": "*",
      "type": "throttlemb",
      "config": {
        "period": "hour",
        "mb": 30,
        "per": 1
      }
    }
  ]
}
```
 it updated the above record which used to be 27. But in ui , i am getting two records upon saving, one is with 27 and another is with 30. Please suggest
**Additional information**

**Comment by cwiechmann** [comment]
@hakila, I'm not sure what you would like to achieve. I mean it make no sense to configure a duplicated quota for one API-Method or all API-Methods. Even if the following is possible in the UI:
![image](https://user-images.githubusercontent.com/16292002/62653689-6390a900-b95e-11e9-8294-bb93390c489f.png)
I'm not sure which impact it will have for an API at runtime. Just guessing, I think one or the other entry is picked up by the API-Manager to enforce the quota. But it will for sure not sum-up the total quota.

The following makes sense:
![image](https://user-images.githubusercontent.com/16292002/62653818-b10d1600-b95e-11e9-8a9a-ad6cea94672a.png)
But I realized this use-case is not yet supported by Swagger-Promote and should be improved.

**Comment by hakila** [comment]
I agree that I don't achieve but found different behaviour with UI and script and hence want to bring it to your notice.

**Comment by cwiechmann** [comment]
Hi @hakila, I made some major improvements for the Quota-Handling which will be part of the upcoming version 1.6.1.
See here for more details: https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/wiki/2.3-Configure-API-Quotas

**Comment by cwiechmann** [comment]
Quota handling has been improved in version 1.6.1

**Comment by hakila** [comment]
I just consumed 1.6.1 and testing quota. I had 37 messages per hour for the api. I put this
{u'restrictions': [{u'config': {u'messages': 38, u'period': u'hours', u'per': 1}, if you notice, i put hours not hour. script simply ignored and did not make any chnages. It is exiting with 0. Could you please check?
------------------------------------------------------------------------
API-Manager Promote Version: 1.6.1

To report issues or get help, please visit:
https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote
------------------------------------------------------------------------

0      INFO  nvironmentProperties| Loaded environment properties from file: env.properties.
784    INFO     APIManagerAdapter| API-Manager version is: 7.7.0
894    INFO  IImportConfigAdapter| Reading API-Definition (Swagger/WSDL) from file: '/opt/axway/pythonscripts/conf/apiswagger/apicreatejson/TESTSWGIMSCRT16SSMinimumStayCategory06Rules_transformed.json' (relative path)
928    INFO  IImportConfigAdapter| Handling configured client-applications.
991    INFO  IImportConfigAdapter| Found existing application: 'ATP_ATP_Apps' based on given name 'ATP_ATP_Apps'
992    INFO  IImportConfigAdapter| Found existing application: 'ATP_CUS_App' based on given name 'ATP_CUS_App'
992    INFO  IImportConfigAdapter| Found existing application: 'X91_91_App' based on given name 'X91_91_App'
992    INFO  IImportConfigAdapter| Found existing application: 'V1S_EY_App' based on given name 'V1S_EY_App'
1084   INFO     APIManagerAdapter| Found existing API on path: '/services/poc/v1/rules/record3/cat6' (published) (ID: 'fbd76029-6620-419f-b73d-cf67620bf4ce')
2375   INFO        APIChangeState| Changed property: clientOrganizations[Desired: '[ATP_ATP_Org, X91_91_Org, V1S_EY_Org, API Development]' vs Actual: '[V1S_EY_Org, ATP_CUS_Org, API Development, ATP_ATP_Org, X91_91_Org]']
2511   INFO        APIChangeState| Changed property: image[Desired: 'APIImage [bytes=9654]' vs Actual: 'APIImage [bytes=13883]']
2511   INFO        APIChangeState| Changed property: applicationQuota[Desired: 'APIQuota [id=null, type=APPLICATION, restrictions=[QuotaRestriction [api=null, method=*, type=throttle, config={messages=38, period=hours, per=1}]]]' vs Actual: 'APIQuota [id=null, type=APPLICATION, restrictions=[QuotaRestriction [api=fbd76029-6620-419f-b73d-cf67620bf4ce, method=*, type=throttle, config={messages=37, period=hour, per=1}], QuotaRestriction [api=fbd76029-6620-419f-b73d-cf67620bf4ce, method=*, type=throttle, config={messages=38, period=hours, per=1}]]]']
2511   INFO     APIManagerAdapter| Strategy: Going to update existing API: TEST SWGIMSCRT16SS Minimum Stay (Category 06) Rules (Version: null)
2511   INFO     APIManagerAdapter| Recognized the following changes. Potentially Breaking: [] plus Non-Breaking: [clientOrganizations, image, applicationQuota]
2511   INFO     APIManagerAdapter| Strategy: Update existing API, as all changes can be applied in current state.
2551   INFO     APIManagerAdapter| Updating API-Image from: /opt/axway/apidocs/markdown/config/resources/images//Ticket.png
2591   INFO     APIManagerAdapter| Updating Application-Default-Quota for API: TEST SWGIMSCRT16SS Minimum Stay (Category 06) Rules
2734   INFO     APIManagerAdapter| Application-Default quota successfully updated: TEST SWGIMSCRT16SS Minimum Stay (Category 06) Rules
2735   INFO     APIManagerAdapter| All desired organizations: [ATP_ATP_Org, X91_91_Org, V1S_EY_Org, API Development] have already access. Nothing to do.
2735   INFO     APIManagerAdapter| NOT removing access for existing orgs: [ATP_CUS_Org] from API: TEST SWGIMSCRT16SS Minimum Stay (Category 06) Rules as clientOrgsMode NOT set to replace.
2736   INFO     APIManagerAdapter| All desired applications: [[ATP_ATP_Apps], [ATP_CUS_App], [X91_91_App], [V1S_EY_App]] have already a subscription. Nothing to do.
2740   INFO                   App| Created API-Properties file: '/opt/axway/pythonscripts/conf/apiswagger/apiconfigjson/TESTSWGIMSCRT16SSMinimumStayCategory06Rules_Details.prop'
2740   INFO                   App| Successfully replicated API-State into API-Manager
return code: 0

**Comment by cwiechmann** [comment]
There is an open issue #109.

Quotas must be configured correct otherwise API-Manager will not realize it.
